.evergreen/scripts/install-uninstall-check-mingw.cmd

```diff
@@ -0,0 +1,180 @@
+REM Supported/used environment variables:
+REM    BSON_ONLY      Whether to build only the BSON library.
+
+rem Ensure Cygwin executables like sh.exe are not in PATH
+rem set PATH=C:\Windows\system32;C:\Windows
+
+echo on
+echo
+
+set SRCROOT=%CD%
+set TAR=C:\cygwin\bin\tar
+
+set version=1.31.0
+set major=1
+
+if "%BSON_ONLY%"=="1" (
+   set BUILD_DIR=%CD%\build-dir-bson
+   set INSTALL_DIR=%CD%\install-dir-bson
+) else (
+   set BUILD_DIR=%CD%\build-dir-mongoc
+   set INSTALL_DIR=%CD%\install-dir-mongoc
+)
+rmdir /S /Q %BUILD_DIR% 2>nul || true
+mkdir %BUILD_DIR% || goto :error
+
+rmdir /S /Q %INSTALL_DIR% 2>nul || true
+mkdir %INSTALL_DIR% || goto :error
+
+set PATH=%PATH%;%INSTALL_DIR%\bin
+
+cd %BUILD_DIR% || goto :error
+
+if "%BSON_ONLY%"=="1" (
+  set BSON_ONLY_OPTION=-DENABLE_MONGOC=OFF
+) else (
+  set BSON_ONLY_OPTION=-DENABLE_MONGOC=ON
+)
+
+rem Build libmongoc, with flags that the downstream R driver mongolite uses
+%CMAKE% -G "Ninja" -DCMAKE_INSTALL_PREFIX=%INSTALL_DIR% -DCMAKE_PREFIX_PATH=%INSTALL_DIR%\lib\cmake %BSON_ONLY_OPTION% .. || goto :error
+%CMAKE% --build . || goto :error
+%CMAKE% --build . --target install || goto :error
+
+REM no kms-message components should be installed
+if exist %INSTALL_DIR%\include\kms_message\kms_message.h (
+   echo kms_message.h found!
+   goto :error
+) else (
+   echo kms_message.h check ok
+)
+if exist %INSTALL_DIR%\lib\libkms_message-static.a (
+   echo libkms_message-static.a found!
+   goto :error
+) else (
+   echo libkms_message-static.a check ok
+)
+if exist %INSTALL_DIR%\lib\cmake\kms_message\kms_message-config.cmake (
+   echo kms_message-config.cmake found!
+   goto :error
+) else (
+   echo kms_message-config.cmake check ok
+)
+
+echo > %INSTALL_DIR%\lib\canary.txt || goto :error
+
+dir %INSTALL_DIR%\share\mongo-c-driver || goto :error
+
+rem CDRIVER-6062: uninstall script may return non-zero exit code despite success.
+%CMAKE% --build . --target uninstall || true
+
+if exist %INSTALL_DIR%\lib\pkgconfig\bson%major%.pc (
+   echo bson%major%.pc found!
+   goto :error
+) else (
+   echo bson%major%.pc check ok
+)
+if not exist %INSTALL_DIR%\lib\canary.txt (
+   echo canary.txt not found!
+   goto :error
+) else (
+   echo canary.txt check ok
+)
+if not exist %INSTALL_DIR%\lib (
+   echo %INSTALL_DIR%\lib not found!
+   goto :error
+) else (
+   echo %INSTALL_DIR%\lib check ok
+)
+if "%BSON_ONLY%" NEQ "1" (
+   if exist %INSTALL_DIR%\lib\pkgconfig\mongoc%major%.pc (
+     echo mongoc%major%.pc found!
+     goto :error
+   ) else (
+     echo mongoc%major%.pc check ok
+   )
+   if exist %INSTALL_DIR%\lib\cmake\mongoc-1.0\mongoc-1.0-config.cmake (
```

Comment:
Not introduced in this PR: but I expect this path, and others, need to be updated to account for C driver 2.0 paths.

I do not consider it a blocker for this PR, since these paths have not changed. But I think these scripts may benefit to also check that these files exist before the uninstall step. That may help avoid checks for files that were never installed.

For consideration: I _think_ these tests can be rewritten in bash. This appears to work: https://gist.github.com/kevinAlbs/4e9ec25bb0c72ebd02ee48dbbfa38560 but I have not tested in Evergreen.